src/crypto.rs

```diff
@@ -123,6 +107,51 @@ fn verify_ring(
     Ok(res.is_ok())
 }
 
+fn verify_ring_es(
+    alg: &dyn signature::VerificationAlgorithm,
+    signature: &str,
+    signing_input: &str,
+    key: Key,
+) -> Result<bool> {
+    let bytes = match key {
+        Key::Pkcs8(bytes) => bytes,
+        _ => {
+            return Err(ErrorKind::InvalidKeyFormat)?;
+        }
+    };
+    verify_ring(alg, signature, signing_input, bytes)
+}
+
+fn verify_ring_rsa(
+    alg: &signature::RsaParameters,
+    signature: &str,
+    signing_input: &str,
+    key: Key,
+) -> Result<bool> {
+    match key {
+        Key::Der(bytes) | Key::Pkcs8(bytes) => verify_ring(alg, signature, signing_input, bytes),
+        Key::ModulusExponent(n, e) => {
+            let signature_bytes = base64::decode_config(signature, base64::URL_SAFE_NO_PAD)?;
+            let message = untrusted::Input::from(signing_input.as_bytes());
+            let modulus = untrusted::Input::from(n);
+            let exponent = untrusted::Input::from(e);
+            let expected_signature = untrusted::Input::from(signature_bytes.as_slice());
+
+            let res = signature::primitive::verify_rsa(
```

Comment:
See the change that uses `RsaPublicKeyComponents` in https://github.com/bluejekyll/trust-dns/pull/845 for how to do this in *ring* 0.16.